outlook/src/taskpane/components/Login/Login.tsx

```diff
@@ -45,76 +68,157 @@ class Login extends React.Component<{}, LoginState> {
         return url.origin;
     }
 
-    login = () => {
-        let sanitizedURL = '';
+    login = async () => {
+        let sanitizedURL = null;
         try {
             sanitizedURL = this.sanitizeDbUrl(this.state.baseURL);
```

Comment:
Somehow feels like all these try/catch blocks are not very elegant.
Could we go for something like:

```js
const sanitizedURL = this.sanitizeDbUrl(this.state.baseURL);
if (!sanitizedURL) {
    // could not get a sanitized URL, ask user to verify it
    this.setState({urlState: UrlState.NotYetChecked, urlError: UrlError.InvalidScheme});
    return;
}
```

This might not be possible/easy depending on the implementation of "sanitizeDbUrl", I let you check.

Reply:
It can crash with `new URL(tmpStr)` but I can capture the exception in `sanitizeDbUrl` and make your proposition